genienlp/tasks/almond_task.py

```diff
@@ -604,6 +604,22 @@ def _make_example(self, parts, dir_name=None, **kwargs):
         return Example.from_raw(example_id, context, question, answer, preprocess=self.preprocess_field, lower=False)
 
 
+def split_text_into_sentences(text, lang):
+    import re
+
+    if lang in ['zh', 'ja', 'ko']:
+        sentences = list(re.findall(u'([^!?。]+[!?。]*)\s?', text, flags=re.U))
+    elif lang in ['en']:
+        sentences = re.sub('(?<!\w\.\w.)(?<![A-Z][a-z]\.)(?<=[\.\!\?])\s|(?:。)', '\n', text).split('\n')
+    else:
+        import nltk
```

Comment:
why not use `nltk` for English as well? Also, it has neural models for sentence boundary detection, does this perform as well as those neural models?

Reply:
I tried nltk's "PunktSentenceTokenizer" and unfortunately it doesn't work well for edge cases.
For example: `I was going there...  But now I'm here, i.e. present`
would be tokenized as `["I was going there...  But now I'm here, i.e.", 'present']` using nltk which is wrong for two reasons.
The regex one splits correctly as: `['I was going there...', " But now I'm here, i.e. present"]`
For my simple use cases, the regex seems to be doing OK but I will take a look at the neural one later too.